### PR TITLE
pacman updates

### LINKED ISSUE #91: Ignore dirs starting with a dot in pacman
Initial problem : ._ mac files impacts Package Manager -> [see this issue on reddit](https://www.reddit.com/r/MiyooMini/comments/uy3iyv/onionos_installer_showing_dot_files_macos/)

Solution : ignore dirs starting with a dot in pacman

2 objectives : 

  - get ride of automatic creation of files from macs
  - be able to hide / desactivate some cores in pac-man (and we add a little sentence in the wiki, something like : )

> deprecated cores are disabled in pac-man by adding a "." prefix

[Discord conversation about it](https://discord.com/channels/529983248114122762/1007796369785897090/1010130219530719292)

### PR DESCRIPTION
**Pull request by introkun**
* fixes #91 (exclude Mac's dot items)
* fix compiler warnings

### CODE REVIEW

File: src/packageManager/packageManager.c [hunk 1]
```diff
@@ -164,17 +163,22 @@ void loadRessources()
             char cShort[MAX_LAYER_NAME_SIZE];
             strcpy(cShort, ep->d_name);
 
-            size_t len = strlen(ep->d_name);
+            const char *file_name = ep->d_name;
+            if (strcmp(file_name, ".") == 0 || strcmp(file_name, "..") == 0)
```
Comment (Aemiii91) [inline src/packageManager/packageManager.c:167]: Why did you add these? It's not necessary when we have line 173.

**Review by Aemiii91 — CHANGES_REQUESTED** [review]
I think we can remove the redundant filename check.

File: src/packageManager/packageManager.c (hunk 1 above)
Comment (introkun) [reply src/packageManager/packageManager.c:167]: Agree

**Review by Aemiii91 — APPROVED** [review]
